src/Sarif/Visitors/SarifVersionOneToCurrentVisitor.cs

```diff
@@ -893,7 +893,6 @@ internal Run CreateRun(RunVersionOne v1Run)
                         BaselineInstanceGuid = v1Run.BaselineId,
                         InstanceGuid = v1Run.Id,
                         Properties = v1Run.Properties,
-                        Results = new List<Result>(),
```

Comment:
Ok, I re-read your PR summary comment. I see that this addresses the "also" -- don't report an empty collection in the v2 output. But:

1. Actually, Michael and I have been discussing the semantic difference between "empty results collection" and "missing results collection". We are settling on an interpretation where "empty" means "I ran the analysis and didn't find any problems", and "missing" means "I couldn't even run the analysis". If we agree with that interpretation, then the v2 file should have a results collection _whenever the v1 file does_ -- and not when it doesn't. Will this change have that effect? Can we write tests to ensure that it works that way?

2. In any case, since this change addresses the "also" in the PR summary comment, my question is: Where is the change that addresses the _title_ of the PR? That is, where is the change that _protects_ against a null v1 results collection?

---
In reply to: [218562462](https://github.com/Microsoft/sarif-sdk/pull/1094#discussion_r218562462) [](ancestors = 218562462)

Reply:
When we were doing the transformer, I remember agreeing that we shouldn't have empty collections. Using its presence with zero items vs. null as an indicator makes sense. I can definitely cover this with tests.
The nullref fix is line 951: if (v1Run.Results != null) {...}

---
In reply to: [218564142](https://github.com/Microsoft/sarif-sdk/pull/1094#discussion_r218564142) [](ancestors = 218564142,218562462)

Reply:
And actually, this change does exactly what we want. If v1Run.Results is empty, we will create an empty v2 results collection. If it's null, we don't create v2 results[].

---
In reply to: [218575678](https://github.com/Microsoft/sarif-sdk/pull/1094#discussion_r218575678) [](ancestors = 218575678,218564142,218562462)

Reply:
My fault, I didn't see the change on Line 951.

Yes, this interpretation of an empty results collection as meaning "I ran the analysis and didn't find any problems" runs counter to our previous agreement to omit empty collections.

---
In reply to: [218576622](https://github.com/Microsoft/sarif-sdk/pull/1094#discussion_r218576622) [](ancestors = 218576622,218575678,218564142,218562462)